### ISSUE TITLE
scoped_pipeline_context emits spurious events to the instance with which it's called

### ISSUE THREAD

**Issue by mgasner** [dagster-io/dagster#3685]
this script fails with an `AssertionError`:
```from dagster import pipeline, solid, DagsterInstance
from dagster.core.execution.api import create_execution_plan, scoped_pipeline_context
@solid
def noop(_):
    pass
@pipeline
def noop_pipeline():
    noop()
instance = DagsterInstance.get()
pipeline_run = instance.create_run_for_pipeline(noop_pipeline)
execution_plan = create_execution_plan(noop_pipeline)
assert instance.all_logs(pipeline_run.run_id) == []
with scoped_pipeline_context(
    execution_plan=execution_plan,
    run_config={},
    pipeline_run=pipeline_run,
    instance=instance,
) as pipeline_context:
    pass
assert instance.all_logs(pipeline_run.run_id) == []```

**Comment by alangenfeld** [comment]
@dpeng817 - i think this might be resolvable with `build_resources` ?

**Comment by dpeng817** [comment]
This is being caused by `scoped_pipeline_context` going back through the initialization process for the context, which emits events. Using `build_resources` to build io managers when retrieving outputs would solve this, but since `intermediate_storage` is still around, we would also need to be able to initialize that. Additionally, without the help of the pipeline context, creating IO manager contexts is very painful.

Tldr; this is something that we _could_ solve, but the solution will be a lot better once intermediate storage is removed, and we have a way of initializing input/output context without the pipeline context.
